OhmGraphite/MetricTimer.cs

```diff
@@ -41,10 +44,11 @@ private async void ReportMetrics(object sender, ElapsedEventArgs e)
                 // every 5 seconds, and there are ways to optimize this, but opening a
                 // new connection is the easiest way to ensure that previous failures
                 // don't affect future results
-                var stopwatch = Stopwatch.StartNew();
-                var sensors = _collector.ReadAllSensors().ToList();
+                _stopwatch.Start();
+                var sensors = _collector.ReadAllSensors();
                 await _writer.ReportMetrics(e.SignalTime, sensors);
-                Logger.Info($"Sent {sensors.Count} metrics in {stopwatch.Elapsed.TotalMilliseconds}ms");
+                Logger.Info($"Sent {sensors.Count()} metrics in {_stopwatch.Elapsed.TotalMilliseconds}ms");
```

Comment:
Since `IEnumerable` is lazy, I believe this will cause the enumerator to be evaluated twice (ie: the sensors will be polled again). There's been issues in the past where polling hardware for sensor values is relatively expensive compared to an intermediate list. Let me know if you've seen otherwise.

IEnumerables being evaluated twice can be seen in the below snippet:

```csharp
using System;
using System.Linq;
                    
public class Program
{
    public static void Main()
    {
        var mylist = new String[] {"Hello", "world", "test"};
        var fives = mylist.Where(x => {
            Console.WriteLine("evaluating {0}", x);
            return x.Length == 5;
        });
        
        foreach (var entry in fives) {
            Console.WriteLine(entry);
        }
        
        Console.WriteLine("wrote {0:d} entries", fives.Count());
    }
}
```

will output

```
evaluating Hello
Hello
evaluating world
world
evaluating test
evaluating Hello
evaluating world
evaluating test
wrote 2 entries
```

Reply:
Sorry, was not aware of that (it's been a long time since I've done C# seriously). I'll close this out.